cdc/changefeed.go

```diff
@@ -188,151 +185,108 @@ func (c *changeFeed) minimumTablesCapture(captures map[string]*model.CaptureInfo
-		if err != nil {
-			return errors.Trace(err)
+	newTaskStatus := make(map[model.CaptureID]*model.TaskStatus, len(captures))
+	cleanedTables := make(map[model.TableID]struct{})
+	addedTables := make(map[model.TableID]struct{})
+
+	for id, targetTs := range c.toCleanTables {
+		captureID, taskStatus, ok := findTaskStatusWithTable(c.taskStatus, id)
+		if !ok {
+			log.Warn("ignore clean table id", zap.Int64("id", id))
+			delete(c.toCleanTables, id)
+			continue
 		}
-		switch lockStatus {
-		case model.TableNoLock:
-			log.Debug("no c-lock", zap.Uint64("tableID", tableID), zap.String("captureID", captureID))
-			delete(c.waitingConfirmTables, tableID)
-		case model.TablePLock:
-			log.Debug("waiting the c-lock", zap.Uint64("tableID", tableID), zap.String("captureID", captureID))
-		case model.TablePLockCommited:
-			log.Debug("delete the c-lock", zap.Uint64("tableID", tableID), zap.String("captureID", captureID))
-			delete(c.waitingConfirmTables, tableID)
+		status, exist := newTaskStatus[captureID]
+		if !exist {
+			status = taskStatus.Clone()
 		}
+		status.RemoveTable(id)
+		status.Operation = append(taskStatus.Operation, &model.TableOperation{
+			TableID:    id,
+			Delete:     true,
+			BoundaryTs: targetTs,
+		})
+		newTaskStatus[captureID] = status
+		cleanedTables[id] = struct{}{}
 	}
 
-	appendTableInfos := make(map[string][]*model.ProcessTableInfo, len(captures))
-
-	for tableID, orphan := range c.orphanTables {
-		captureID := c.selectCapture(captures, appendTableInfos)
+	for tableID, startTs := range c.orphanTables {
+		captureID := c.selectCapture(captures, newTaskStatus)
 		if len(captureID) == 0 {
 			return nil
 		}
-		info := appendTableInfos[captureID]
-		info = append(info, &model.ProcessTableInfo{
-			ID:      tableID,
-			StartTs: orphan.StartTs,
+		taskStatus := c.taskStatus[captureID]
+		if taskStatus == nil {
+			taskStatus = new(model.TaskStatus)
+		}
+		status, exist := newTaskStatus[captureID]
+		if !exist {
+			status = taskStatus.Clone()
+		}
+		status.Tables[tableID] = startTs
+		status.Operation = append(taskStatus.Operation, &model.TableOperation{
+			TableID:    tableID,
+			BoundaryTs: startTs,
 		})
-		appendTableInfos[captureID] = info
+		newTaskStatus[captureID] = status
+		addedTables[tableID] = struct{}{}
 	}
 
-	for captureID, tableInfos := range appendTableInfos {
-		status := c.taskStatus[captureID]
-		if status == nil {
-			status = new(model.TaskStatus)
-		}
-		statusClone := status.Clone()
-		status.TableInfos = append(status.TableInfos, tableInfos...)
-
-		newInfo, err := c.infoWriter.Write(ctx, c.id, captureID, status, true)
-		if err == nil {
-			c.taskStatus[captureID] = newInfo
-		}
-		switch errors.Cause(err) {
-		case model.ErrFindPLockNotCommit:
-			c.restoreTableInfos(statusClone, captureID)
-			log.Info("write table info delay, wait plock resolve",
-				zap.String("changefeed", c.id),
-				zap.String("capture", captureID))
-		case nil:
-			log.Info("dispatch table success",
```

Comment:
we can add some logs like this to record `table dispatch` history, which is often useful to diagnosis